### ISSUE TITLE
Add support for ADMIN_USER_PASSWORD_AUTH

### ISSUE THREAD

**Issue by franz-see** [spulec/moto#2663]
> ADMIN_USER_PASSWORD_AUTH: Admin-based user password authentication. This replaces the ADMIN_NO_SRP_AUTH authentication flow. In this flow, Cognito receives the password in the request instead of using the SRP process to verify passwords.

https://docs.aws.amazon.com/cognito-user-identity-pools/latest/APIReference/API_AdminInitiateAuth.html#API_AdminInitiateAuth_RequestSyntax

So i think doing this change should suffice

```
diff --git a/moto/cognitoidp/models.py b/moto/cognitoidp/models.py
index 6700920c..cf10e8d0 100644
--- a/moto/cognitoidp/models.py
+++ b/moto/cognitoidp/models.py
@@ -644,7 +644,7 @@ class CognitoIdpBackend(BaseBackend):
         if not client:
             raise ResourceNotFoundError(client_id)
 
-        if auth_flow == "ADMIN_NO_SRP_AUTH":
+        if auth_flow == "ADMIN_NO_SRP_AUTH" or auth_flow == "ADMIN_USER_PASSWORD_AUTH":
             username = auth_parameters.get("USERNAME")
             password = auth_parameters.get("PASSWORD")
             user = user_pool.users.get(username)
```

**Comment by franz-see** [comment]
Pull Request to fix this one - https://github.com/spulec/moto/pull/2664

**Comment by franz-see** [comment]
@gruebel this one does not have formatting errors :)

**Comment by bblommers** [comment]
This is now fixed as of moto >= 1.3.16.dev184
